app/models/concerns/orchestration/dhcp.rb

```diff
@@ -164,8 +164,13 @@ def queue_dhcp_update
   # do we need to update our dhcp reservations
   def dhcp_update_required?
     # IP Address / name changed, or 'rebuild' action is triggered and DHCP record on the smart proxy is not present/identical.
-    return true if ((old.ip != ip) || (old.hostname != hostname) || provision_mac_addresses_changed? || (old.subnet != subnet) || (operatingsystem.boot_filename(old.host) != operatingsystem.boot_filename(self.host)) ||
-                    (!old.build? && build? && !all_dhcp_records_valid?))
+    return true if ((old.ip != ip) ||
```

Comment:
This is getting very long. I'm wondering if we should have some helper:

```ruby
attributes = [
  [old.ip, ip],
  [old.hostname, hostname],
  [old.subnet, subnet],
  [old.mac_addresses_for_provisioning, mac_addresses_for_provisioning],
  # ...
]

return true if attributes.any? { |old_value, new_value| old_value != new_value }
```

If calls do additional database access or API calls then it's fine to add additional lines with `return true`.

Reply:
I can format it more nicely, but what you suggest is worse readability. Also and what's more important is that your code evaluates all statements at once and while this might be not a problem now, it can break in the future. Or people can start adding if-statements back when there will be dependencies.

Reply:
I didn't write out the full version, but a more optimal version that I don't exactly know the Ruby equivalent for so bear with me:
```python
attributes = [
  :ip,
  :hostname,
  :subnet,
  :mac_addresses_for_provisioning,
  # ...
]
return true if attributes.any? { |attr| old.send(attr) != send(attr) }
```

If there are more complex statements, you can easily add them after the return true. That's why there's a return true instead of just `attributes.any? { ... }`. My point is that there's a simple list of attributes in a specific order we compare on both objects and if they differ, we must trigger an update. We can be pretty specific about those attributes. AFAIK `send` actually calls methods so you can add a helper as well.

Then when you want to debug, you can actually easily spit out a table to compare the values manually as well.